src/mlir/verona-mlir.cc

```diff
@@ -14,107 +13,111 @@
+  void cmdLineDefaults()
   {
-    CLI::App app{"Verona MLIR"};
-
-    Opt opt;
-    app.add_flag("--emit-mlir", opt.mlir, "Emit MLIR.");
-    app.add_flag("--emit-llvm", opt.llvm, "Emit LLVM (default).");
-    app.add_option("-g,--grammar", opt.grammar, "Grammar to use.");
-    app.add_option("-o,--output", opt.output, "Output filename.");
-    app.add_option("file", opt.filename, "File to compile.");
+    // Default input is stdin
+    if (inputFile.empty())
+      inputFile = "-";
 
-    try
-    {
-      app.parse(argc, argv);
-    }
-    catch (const CLI::ParseError& e)
+    // Detect source type from extension, if not passed as argument
+    if (inputKind == InputKind::None)
     {
-      exit(app.exit(e));
+      llvm::StringRef filename(inputFile);
+      if (filename.endswith(".verona"))
+        inputKind = InputKind::Verona;
+      else if (filename.endswith(".mlir"))
+        inputKind = InputKind::MLIR;
+      else if (filename == "-") // STDIN, assume Verona
+        inputKind = InputKind::Verona;
     }
 
-    // Default is to output MLIR
-    if (!opt.llvm)
-      opt.mlir = true;
-
-    // Default grammar
-    if (opt.grammar.empty())
-      opt.grammar = path::directory(path::executable()).append("/grammar.peg");
-
-    // Default input is stdin
-    if (opt.filename.empty())
-      opt.filename = "-";
-
-    return opt;
-  }
-
-  // Print help
-  void help()
-  {
-    std::cout << "Compiler Syntax: verona-mlir AST|MLIR|LLVM <filename.verona>"
-              << std::endl;
-  }
-
-  // Detect source type from extension
-  SourceKind getSourceType(llvm::StringRef filename)
-  {
-    auto source = SourceKind::None;
-    if (filename.endswith(".verona"))
-      source = SourceKind::Verona;
-    else if (filename.endswith(".mlir"))
-      source = SourceKind::MLIR;
-    else if (filename == "-") // STDIN, assume MLIR
-      source = SourceKind::MLIR;
-    return source;
-  }
-
-  // Choose output file extension from output type
-  // Careful with mlir->mlir not to overwrite source file
-  std::string
-  getOutputFilename(llvm::StringRef filename, Opt& opt, SourceKind source)
-  {
-    if (!opt.output.empty())
-      return opt.output;
-    if (filename == "-")
-      return "-";
+    // Default to output MLIR
+    if (outputKind == OutputKind::None)
+      outputKind = OutputKind::MLIR;
 
-    std::string newName = filename.substr(0, filename.find_last_of('.')).str();
-    if (opt.mlir)
+    // Choose output file extension from output type
+    // Careful with mlir->mlir not to overwrite source file
+    if (outputFile.empty())
     {
-      if (source == SourceKind::MLIR)
-        newName += ".mlir.out";
+      llvm::StringRef filename(inputFile);
+      if (filename == "-")
+      {
+        outputFile = "-";
+      }
       else
-        newName += ".mlir";
-    }
-    else
-    {
-      newName += ".ll";
+      {
+        std::string newName =
+          filename.substr(0, filename.find_last_of('.')).str();
```

Comment:
I don't really like the fact that we're using strings for paths here.  Can we not use `llvm::sys::path` and its `replace_extension` method?

Reply:
Good point, will do.